src/kaleidoscope/hardware/softhruf/Splitography.h

```diff
@@ -26,33 +26,34 @@
 
 #ifdef ARDUINO_AVR_SPLITOGRAPHY
 
+#define KALEIDOSCOPE_BOOTLOADER_FLIP 1
+
 #include <Arduino.h>
 #define HARDWARE_IMPLEMENTATION kaleidoscope::hardware::softhruf::Splitography
-#include "Kaleidoscope-HIDAdaptor-KeyboardioHID.h"
-
-#include "kaleidoscope/macro_helpers.h"
-#include "kaleidoscope/hardware/avr/pins_and_ports.h"
 
-#include "kaleidoscope/hardware/ATMegaKeyboard.h"
+#include "kaleidoscope/driver/keyscanner/ATMegaKeyScanner.h"
+#include "kaleidoscope/driver/bootloader/avr/FLIP.h"
+#include "kaleidoscope/hardware/avr/AVRDevice.h"
 
 namespace kaleidoscope {
 namespace hardware {
 namespace softhruf {
-class Splitography: public kaleidoscope::hardware::ATMegaKeyboard {
- public:
-  Splitography(void) {
-    mcu_.disableJTAG();
-  }
 
-  ATMEGA_KEYBOARD_CONFIG(
-    ROW_PIN_LIST({ PIN_D0, PIN_D1, PIN_D2, PIN_D3 }),
-    COL_PIN_LIST({ PIN_F0, PIN_F1, PIN_F4, PIN_F5, PIN_F6, PIN_F7, PIN_C7, PIN_C6, PIN_B6, PIN_B5, PIN_B4, PIN_D7 })
-  );
-
-  static constexpr int8_t led_count = 0;
+struct SplitographyDeviceBlueprint : kaleidoscope::hardware::avr::AVRDeviceBlueprint {
+  typedef struct SplitographyKeyScannerBlueprint : public kaleidoscope::driver::keyscanner::ATMegaKeyScannerBlueprint {
+    ATMEGA_KEYSCANNER_BLUEPRINT(
+      ROW_PIN_LIST({ PIN_D0, PIN_D1, PIN_D2, PIN_D3 }),
+      COL_PIN_LIST({ PIN_F0, PIN_F1, PIN_F4, PIN_F5, PIN_F6, PIN_F7, PIN_C7, PIN_C6, PIN_B6, PIN_B5, PIN_B4, PIN_D7 })
+    );
+  } KeyScannerBlueprint;
+  typedef kaleidoscope::driver::keyscanner::ATMegaKeyScanner<KeyScannerBlueprint> KeyScanner;
+  typedef kaleidoscope::driver::bootloader::avr::FLIP BootLoader;
+};
 
-  static constexpr int8_t numKeys() {
-    return matrix_columns * matrix_rows;
+class Splitography: public kaleidoscope::hardware::avr::AVRDevice<SplitographyDeviceBlueprint> {
+ public:
+  Splitography() {
+    mcu_.disableJTAG();
```

Comment:
I suspect that disableJTAG() can get lifted to the generic 32u4 class

Reply:
As in, should the generic 32u4 class do the JTAG disable in its constructor, for all boards that use it?

Reply:
Oh. the code as I read it before suggested that the implementation was inside splitography. if this is just a call out to disableJTAG defined generically, that's fine.

Reply:
It is a call out to `ATMega32U4::disableJTAG()`, yup.